orquesta/tests/unit/conducting/test_workflow_conductor_rerun.py

```diff
@@ -0,0 +1,629 @@
+        self.assertDictEqual(conductor.get_task_transition_contexts('task2', 0), {})
+        self.assert_next_task(conductor, has_next_task=False)
+
+        # Check the list of terminal tasks.
+        term_tasks = conductor.workflow_state.get_terminal_tasks()
+        actual_term_tasks = [t['id'] for t in term_tasks]
+        expected_term_tasks = ['task1', 'task3', 'task4', 'task2']
+        self.assertListEqual(actual_term_tasks, expected_term_tasks)
+
+        # Check workflow status, context and output.
+        expected_term_ctx = {'var1': 'xyz', 'var2': 123}
+        self.assertDictEqual(conductor.get_workflow_terminal_context(), expected_term_ctx)
+        self.assertDictEqual(conductor.get_workflow_output(), expected_output)
+        self.assertEqual(conductor.get_workflow_status(), statuses.SUCCEEDED)
+
+    def test_parallel_rerun_with_last_task(self):
+        expected_output = {'var1': 'xyz', 'var2': 123}
+
+        # 1. Prepare rerun workflow
+        conductor = self.pre_rerun_workflow(PARALLEL_WF_DEF)
+
+        # 2. Fail task4
+        # Conduct tasks.
+        self.forward_task_statuses(conductor, 'task1', [statuses.RUNNING])
+        self.forward_task_statuses(conductor, 'task3', [statuses.RUNNING])
+
+        self.forward_task_statuses(conductor, 'task1', [statuses.SUCCEEDED])
+        self.forward_task_statuses(conductor, 'task3', [statuses.SUCCEEDED])
+
+        self.forward_task_statuses(conductor, 'task2', [statuses.RUNNING])
+        self.forward_task_statuses(conductor, 'task4', [statuses.RUNNING])
+
+        self.forward_task_statuses(conductor, 'task2', [statuses.SUCCEEDED])
+        self.forward_task_statuses(conductor, 'task4', [statuses.FAILED])
+
+        # Log the errors
+        errors = ['error1', 'error2']
+        conductor.log_errors(errors, 'task4')
+
+        # Check the list of terminal tasks.
+        term_tasks = conductor.workflow_state.get_terminal_tasks()
+        actual_term_tasks = [t['id'] for t in term_tasks]
+        expected_term_tasks = ['task2', 'task4']
+        self.assertListEqual(actual_term_tasks, expected_term_tasks)
+        self.assertEqual(conductor.get_workflow_status(), statuses.FAILED)
+
+        # 3. Start rerun workflow
+        rerun_tasks = ['task4']
+        number_tasks = 4
+        self.assert_workflow_rerun_request(conductor, rerun_tasks, number_tasks)
+
+        # Conduct task4 and check context.
+        self.forward_task_statuses(conductor, 'task4', [statuses.RUNNING])
+        self.forward_task_statuses(conductor, 'task4', [statuses.SUCCEEDED])
+        self.assertDictEqual(conductor.get_task_transition_contexts('task4', 0), {})
+        self.assert_next_task(conductor, has_next_task=False)
+
+        # Check the list of terminal tasks.
+        term_tasks = conductor.workflow_state.get_terminal_tasks()
+        actual_term_tasks = [t['id'] for t in term_tasks]
+        expected_term_tasks = ['task2', 'task4', 'task4']
```

Comment:
We missed this. The term tasks should only be `['task2', 'task4']`.  In `request_workflow_rerun`, you have to set the original `task['term'] = False`.